notification/app/notification/services/guardian/ReportTopicRegistrationCounter.scala

```diff
@@ -1,27 +1,29 @@
 package notification.services.guardian
-import models.{PlatformCount, Topic}
-import play.api.libs.ws.WSClient
+
+import models.{PlatformCount, Topic, TopicCount}
+import notification.data.DataStore
+import play.api.libs.json.Format
 import utils.LruCache
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration.DurationLong
 
-class ReportTopicRegistrationCounter(
-  ws: WSClient,
-  reportUrl: String,
-  apiKey: String
-)(implicit ec: ExecutionContext) extends TopicRegistrationCounter {
+class ReportTopicRegistrationCounter(topicCountDataStore: DataStore[TopicCount])(implicit ec: ExecutionContext) extends TopicRegistrationCounter {
 
   val lruCache: LruCache[PlatformCount] = new LruCache[PlatformCount](200, 1000, 3.days)
 
-  override def count(topics: List[Topic]): Future[PlatformCount] = {
+  override def count(topics: List[Topic])(implicit format: Format[TopicCount] ): Future[PlatformCount]  =  {
     lruCache(topics.toSet) {
-      val topicParameters = topics.map(topic => "topics" -> topic.toString)
-      ws.url(s"$reportUrl/registration-count")
-        .withQueryStringParameters(topicParameters: _*)
-        .withHttpHeaders("Authorization" -> s"Bearer $apiKey")
-        .get
-        .map(response => response.json.as[PlatformCount])
+      val topicNames = topics.map(topic => topic.fullName)
+      val persitedTopicRegistrationCounts = topicCountDataStore.get()
+      persitedTopicRegistrationCounts.map {
+        top =>
+          val totalRegistrationsForTopics = top.filter{
+            topicCount => topicNames.contains(topicCount.topicName)
+          }
+          .foldRight(0){ (topic, count) => count + topic.registrationCount }
```

Comment:
Is `.map(_.registrationCount).sum` easier?

Reply:
Yes